VariantShredding.md

```diff
@@ -25,276 +25,302 @@
 
-Below is a sample of JSON that would be fully shredded in this example.
-It contains an array of objects, containing an `a` field shredded as an array, and a `b` field shredded as an integer.
+The series of objects below would be stored as:
 
-```
-[
-  {
-    "a": [1, 2, 3],
-    "b": 100
-  },
-  {
-    "a": [4, 5, 6],
-    "b": 200
-  }
-]
-```
+| Event object                                                                       | `value`                           | `typed_value` | `typed_value.event_type.value` | `typed_value.event_type.typed_value` | `typed_value.event_ts.value` | `typed_value.event_ts.typed_value` | Notes                                         |
+|------------------------------------------------------------------------------------|-----------------------------------|---------------|--------------------------------|--------------------------------------|------------------------------|------------------------------------|-----------------------------------------------|
+| `{"event_type": "noop", "event_ts": 1729794114937}`                                | null                              | non-null      | null                           | `noop`                               | null                         | 1729794114937                      | Fully shredded object                         |
+| `{"event_type": "login", "event_ts": 1729794146402, "email": "user@example.com"}`  | `{"email": "user@example.com"}`   | non-null      | null                           | `login`                              | null                         | 1729794146402                      | Partially shredded object                     |
+| `{"error_msg": "malformed: ..."}`                                                  | `{"error_msg", "malformed: ..."}` | null          |                                |                                      |                              |                                    | Object with no shredding                      |
+| `"malformed: not an object"`                                                       | `malformed: not an object`        | null          |                                |                                      |                              |                                    | Not an object (stored as Variant string)      |
+| `{"event_ts": 1729794240241, "click": "_button"}`                                  | `{"click": "_button"}`            | non-null      | null                           | null                                 | null                         | 1729794240241                      | Field `event_type` is missing                 |
+| `{"event_type": null, "event_ts": 1729794954163}`                                  | null                              | non-null      | `00` (field exists, is null)   | null                                 | null                         | 1729794954163                      | Field `event_type` is present and is null     |
```

Comment:
Some more requested examples, 

Could we have where "event_ts" is a Date or something non transformable into a timestamp?
I assume this would make value be `{"event_ts": "08-03-2025"}` while typed_value would be `null`

I also wonder if we could do a single example for a doubly nested field showing where typed_value.address.value != null. All the examples here cover a primitive field being typed, so It may be nice to show the behavior with a object being typed. 
```
{
 Name
 Address {
    City 
    ZIP (Shredded as INT but some values as String?)
    }
}
```

Reply:
I added most other examples, but I don't think that we need the nested example because it would make the table much larger. I also cover nesting in the next section specifically.